lib/curly/component_compiler.rb

```diff
@@ -90,7 +89,7 @@ def validate_block_argument!
 
     def validate_attributes!
       attributes.keys.each do |key|
-        unless attribute_names.include?(key)
+        unless attribute_names.include?(key) || attributes_collected?
```

Comment:
In the case where `attributes_collected? == true` there's no need to iterate through `attributes` – maybe just return early at the top of the method then?